scripts/fetch_data.py

```diff
@@ -0,0 +1,133 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import logging
+import os
+import subprocess
+import time
+
+LFS_DATASETS = {
+    'all': '**',
+    'examples': 'examples/**',
+    'models': 'models/**',
+    'tests': 'tests/**',
+    'validation': 'models/datasets/validation-data/**'
+}
+
+
+def lfsPull(include_paths, poll_interval=0.1):
+    """
+    Performs a git lfs pull.
+    """
+    cmd = 'git lfs pull -I "{}"'.format(','.join(include_paths))
+    env = os.environ.copy()
+
+    # Instruct git lfs to not supress progress output. Fetching the models can
+    # take over a minute to complete, so we want our users to receive feedback.
+    env['GIT_LFS_FORCE_PROGRESS'] = '1'
+    popen = subprocess.Popen(cmd,
+                             env=env,
+                             shell=True,
+                             universal_newlines=True,
+                             stderr=subprocess.STDOUT,
+                             stdout=subprocess.PIPE)
+
+    outpipe = popen.stdout
+    returncode = None
+    all_out = []
+    while returncode is None:
+        time.sleep(poll_interval)
+        out = outpipe.readline()
+        if out.rstrip() != '':
+            logging.info(out.rstrip())
+            all_out.append(out)
+
+        returncode = popen.poll()
+
+    # Check if we have any additional output written to the pipe before our last poll
+    out = outpipe.read()
+    if out != '':
+        all_out.append(out)
+
+    output = ''.join(all_out).rstrip("\n")
+    if returncode != 0:
+        logging.error(output)
+        raise subprocess.CalledProcessError(returncode=returncode, cmd=cmd, output=output)
+
+    return output
+
+
+def lfsCheck(list_all=False):
+    output = subprocess.check_output('git lfs ls-files', shell=True, universal_newlines=True)
+    output_lines = output.splitlines()
+
+    # Output lines are in the format of:
+    # <oid> [-|*] <file name>
+    # where '-' indicates a file pointer and '*' indicates a downloaded file
+    # https://github.com/git-lfs/git-lfs/blob/main/docs/man/git-lfs-ls-files.1.ronn
+
+    all_downloaded = True
+    for file_status in output_lines:
+        parts = file_status.split()
+        downloaded = parts[1] == '*'
+        if all_downloaded and not downloaded:
+            all_downloaded = False
+
+        if list_all:
+            # the join on 2: is needed to handle file names that contain a blank space
+            logging.info('%s - %s', ' '.join(parts[2:]), downloaded)
+
+    if not list_all:
+        logging.info(all_downloaded)
```

Comment:
I ran this and it was not clear at all to me what True/False meant. Can this instead return 0/1 error code (incase this script is used in CI or something) and print something like: "All LFS files downloaded" on true and "Missing the following LFS files: <list of missing files>" on false? Keep the `--list` option the way it currently is.